### PR TITLE
Rust errs

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: src/rust/bitbox02/src/memory.rs [hunk 1]
```diff
@@ -16,16 +16,18 @@
 // deduct one for the null terminator.
 pub const DEVICE_NAME_MAX_LEN: usize = bitbox02_sys::MEMORY_DEVICE_NAME_MAX_LEN as usize - 1;
 
+pub struct Error {}
```
Comment (x1ddos) [inline src/rust/bitbox02/src/memory.rs:19]: Can just be a unit struct? There's nothing to store in it.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9e3e97d945e0d3ada821802c00b12e13

**Review by x1ddos — APPROVED** [review]
> Only one function is converted for now. Can convert the rest as well.

Please do! This is nice.

File: src/rust/bitbox02-rust/src/hww/api/backup.rs [hunk 2]
```diff
@@ -87,7 +87,7 @@ pub async fn create(
     const MAX_WEST_UTC_OFFSET: i32 = -43200; // 12 hours in seconds
 
     if timezone_offset < MAX_WEST_UTC_OFFSET || timezone_offset > MAX_EAST_UTC_OFFSET {
-        return Err(Error::COMMANDER_ERR_INVALID_INPUT);
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/backup.rs:90]: it already is, the section in this diff is red.

File: src/rust/bitbox02/src/memory.rs (hunk 1 above)
Comment (benma) [reply src/rust/bitbox02/src/memory.rs:19]: oh cool didn't know. done

File: src/rust/bitbox02-rust/src/hww/api/error.rs [hunk 3]
```diff
@@ -63,3 +63,9 @@ pub fn make_error(err: Error) -> Response {
     };
     Response::Error(err)
 }
+
+impl core::convert::From<()> for Error {
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/error.rs:67]: :+1: done

File: src/rust/bitbox02-rust/src/hww/api/backup.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (x1ddos) [reply src/rust/bitbox02-rust/src/hww/api/backup.rs:90]: oh, it's in a separate commit. I see. sorry.

**Review by x1ddos — APPROVED** [review]
LGTM

**Pull request merged**